backend/lib/endpoints/reports.js

```diff
@@ -40,15 +48,599 @@ async function routes(app) {
         throw app.httpErrors.notFound();
       }
 
-      // action, post, actorId (triggredBy), authUserId, details
-      //app.notifier.notify("report", updatedPost, actor._id, userId);
+      reply.code(201);
+      return {
+        success: true,
+      };
+    },
+  );
+
+  // get all reported posts
+  app.get(
+    "/posts",
+    {
+      preValidation: [
+        app.authenticate,
+        app.setActor,
+        app.checkScopes([SCOPES.REPORT_READ]),
+      ],
+      schema: getPostReportsSchema,
+    },
+    async (req) => {
+      const {
+        query: { limit, status, skip, includeMeta, keywords },
+      } = req;
+
+      const filters = [{ status: status }];
+      if (status === "public") {
+        // rejected report, status is "public", but have reportedBy.length
+        filters.push({ reportedBy: { $exists: true, $not: { $size: 0 } } });
+      }
+
+      const matchingStage = {
+        $match: { $and: filters },
+      };
+      const sortingStage = {};
+
+      if (keywords) {
+        matchingStage.$match.$text = {
+          $search: keywords,
+          $caseSensitive: false,
+          $diacriticSensitive: false,
+        };
+        sortingStage.$sort = { score: { $meta: "textScore" } };
+      } else {
+        sortingStage.$sort = {
+          [status === "flagged" ? "reportsCount" : "updatedAt"]: -1,
+        };
+      }
+
+      const aggregationPipelineResults = [
+        matchingStage,
+        {
+          $set: {
+            reportsCount: {
+              $size: {
+                $ifNull: ["$reportedBy", []],
+              },
+            },
+            content: {
+              $concat: [
+                { $substrCP: ["$content", 0, UNLOGGED_POST_SIZE] },
+                "...",
+              ],
+            },
+          },
+        },
+        {
+          $project: {
+            "author.location": false,
+          },
+        },
+        sortingStage,
+        {
+          $skip: parseInt(skip) || 0,
+        },
+        {
+          $limit: Math.min(limit || MAX_REPORTS_PER_PAGE),
+        },
+      ];
+
+      // Get the total results without pagination steps but with filtering aplyed - totalResults
+      /* eslint-disable sort-keys */
+      const totalResultsAggregationPipeline = await Post.aggregate([
+        {
+          $match: { $and: filters },
+        },
+        { $group: { _id: null, count: { $sum: 1 } } },
+      ]);
+      /* eslint-enable sort-keys */
+
+      const [postsErr, posts] = await app.to(
+        Post.aggregate(aggregationPipelineResults).then((posts) => {
+          posts.forEach((post) => {
+            post.elapsedTimeText = {
+              created: translateISOtoRelativeTime(post.createdAt),
```

Comment:
maybe simply remove the frontend duplicated code if that is not used

Reply:
Comments and posts get the time converted from the backend. but when making messaging, I had to generate the time on the front-end (to be relative to the client timezone), so the front-end code is used for messaging and notifications only, and I agree that it should be used for everything else, because it is relative to the client timezone, and not the server's one.
but since this includes changing it for comments and posts not only on the dashboard, I think it should be handled as a separate issue, to not make this PR larger than it is